java/src/test/java/gherkin/GenerateAstTest.java

```diff
@@ -0,0 +1,69 @@
+package gherkin;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+
+public class GenerateAstTest {
+    @Test
+    public void parse_all_good_features() throws IOException {
+        String[] args = getAllGoodTestFeatures();
+
+        GenerateAst.main(args);
```

Comment:
I'm not a big fan of unit tests that print to STDOUT. I'd prefer to refactor `GenerateAst` to avoid this.